runtest-moduleapi

```diff
@@ -13,7 +13,7 @@ then
     exit 1
 fi
 
-$MAKE -C tests/modules && \
+$MAKE -C tests/modules CFLAGS='-Werror' && \
```

Comment:
@enjoy-binbin i think i rather not hard code `Werror` in the makefiles and test files, but only have it enabled in our CI via the yaml files.

we should some day rearrange the build of the module tests in some way so they can take flags from the redis build, or get explicit flags from outside.

if we can't do that now, let's just merge the other improvements in this PR, and leave that part out for some future day.

Reply:
Yes, I think so too, but I haven’t thought of a better way to deal with it